### ISSUE TITLE
Postgresql persistence through YesSql throws exceptions

### ISSUE THREAD

**Issue by yuanyc-cityfun** [elsa-workflows/elsa-core#751]
The Same Problem as #257 When I was use YesSQL PostgresSQL Persistence

pg version: 9 and  10 with postgis
asp.net core 3.1

```
Npgsql.PostgresException (0x80004005): 42883: 操作符不存在(operator does not exist): character varying = integer
   at Npgsql.NpgsqlConnector.<ReadMessage>g__ReadMessageLong|194_0(NpgsqlConnector connector, Boolean async, DataRowLoadingMode dataRowLoadingMode, Boolean readingNotifications, Boolean isReadingPrependedMessage)
   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
   at Dapper.SqlMapper.QueryAsync[T](IDbConnection cnn, Type effectiveType, CommandDefinition command) in C:\projects\dapper\Dapper\SqlMapper.Async.cs:line 419
   at YesSql.Store.ProduceAsync[T](WorkerQueryKey key, Func`2 work, Object[] args) in C:\projects\yessql-un1yf\src\YesSql.Core\Store.cs:line 355
   at YesSql.Services.DefaultQuery.Query`1.ListImpl() in C:\projects\yessql-un1yf\src\YesSql.Core\Services\DefaultQuery.cs:line 1020
   at Elsa.Persistence.YesSql.Services.YesSqlWorkflowInstanceStore.ListByBlockingActivityAsync(String activityType, String correlationId, CancellationToken cancellationToken)
   at Elsa.Extensions.AsyncEnumerableExtensions.ToListAsync[T](Task`1 task)
   at Elsa.Services.WorkflowInvoker.ResumeManyAsync(String activityType, Variables input, String correlationId, Func`2 activityStatePredicate, CancellationToken cancellationToken)
   at Elsa.Services.WorkflowInvoker.TriggerAsync(String activityType, Variables input, String correlationId, Func`2 activityStatePredicate, CancellationToken cancellationToken)
   at NetModular.Module.GeoCMS.Application.ResBindingService.ResBindingService.Submit(Guid id) in D:\Code\GeoCMS\src\Library\Application\ResBindingService\ResBindingService.cs:line 259
   at lambda_method(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.AwaitableObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Logged|12_1(ControllerActionInvoker invoker)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.MiddlewareAnalysis.AnalysisMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.MiddlewareAnalysis.AnalysisMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.MiddlewareAnalysis.AnalysisMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.MiddlewareAnalysis.AnalysisMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.MiddlewareAnalysis.AnalysisMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.MiddlewareAnalysis.AnalysisMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.MiddlewareAnalysis.AnalysisMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.MiddlewareAnalysis.AnalysisMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.MiddlewareAnalysis.AnalysisMiddleware.Invoke(HttpContext httpContext)
  Exception data:
    Severity: 错误
    SqlState: 42883
    MessageText: 操作符不存在: character varying = integer
    Hint: 没有匹配指定名称和参数类型的操作符. 您也许需要增加明确的类型转换.
    Position: 296
    File: parse_oper.c
    Line: 726
```

**Comment by sfmskywalker** [comment]
Given the message `operator does not exist: character varying = integer`, I think it might have something to do with enum field types.

**Comment by sfmskywalker** [comment]
Which means it's most likely not of your doing of course, but a bug in Elsa or YesSQL. Perhaps the fix is as simple as upgrading to the latest YesSQL version. Perhaps this is something you could try?

**Comment by yuanyc-cityfun** [comment]
I tried to upgrade YesSQL but failed, because of  legacy problems in my application.
I found out the reason is because when YesSQL creating tables and columns in Postgressql, enum type is created as varchar(255) .
I changed ‘WorkflowStatus’  to int4(32), and the problem is fixed.

**Comment by craigfowler** [comment]
@yuanyc-cityfun wrote:
> I found out the reason is because when YesSQL creating tables and columns in Postgressql, enum type is created as varchar(255). I changed ‘WorkflowStatus’ to int4(32), and the problem is fixed.

@sfmskywalker I don't really know YesSQL very well, but [is that because of this line in the migrations]?  Should that be a numeric `DbType` perhaps, for storing enums as their integer values rather than `ToString()` equivalent?  If so, easy & quick fix 😎 

[is that because of this line in the migrations]: https://github.com/elsa-workflows/elsa-core/blob/0c15c94ea7d982392b8e0934d9246e826b1525f7/src/persistence/Elsa.Persistence.YesSql/Migrations.cs#L32

**Comment by sfmskywalker** [comment]
I think that may very well be the case, yes. When I did, things got broken, so I complained to [Sebastien](https://github.com/sebastienros) and he fixes YesSQL with support for enum types. So it should work when all the versions align.

**Comment by craigfowler** [comment]
Ahahahaha.  That's the best dev approach I've seen yet.  I'm stealing that one for the day job 😁

**Comment by craigfowler** [comment]
Fixed in Elsa 2.0!
